pkg/worker/verifier/validator_status.go

```diff
@@ -62,26 +65,21 @@ func (cs *ValidatorStatus) ValidateRecord(r *kgo.Record, validRanges *TopicOffse
 			log.Infof("Ignoring read validation at offset outside valid range %s/%d %d", r.Topic, r.Partition, r.Offset)
 		}
 	} else {
-		cs.ValidReads += 1
-		log.Debugf("Read OK (%s) on p=%d at o=%d", r.Headers[0].Value, r.Partition, r.Offset)
-
-		if cs.MaxOffsetsConsumed == nil {
-			cs.MaxOffsetsConsumed = make(map[int32]int64)
-		}
-
-		currentMax, present := cs.MaxOffsetsConsumed[r.Partition]
+		currentMax, present := cs.lastOffsetConsumed[r.Partition]
 		if present {
 			if currentMax < r.Offset {
 				expected := currentMax + 1
 				if r.Offset != expected {
 					log.Warnf("Gap detected in consumed offsets. Expected %d, but got %d", expected, r.Offset)
```

Comment:
nit: Gaps are not ok for sequential readers if the partition is not compacted, but I guess we don't have enough information to determine that here.